storage/shard.go

```diff
@@ -1467,7 +1467,10 @@ func (s *dbShard) FetchBlocksMetadataV2(
 	// Work backwards while in requested range and not before retention
 	for !blockStart.Before(start) &&
 		!blockStart.Before(retention.FlushTimeStart(ropts, s.nowFn())) {
-		exists := s.namespaceReaderMgr.filesetExistsAt(s.shard, blockStart)
+		exists, err := s.namespaceReaderMgr.filesetExistsAt(s.shard, blockStart)
+		if err != nil {
```

Comment:
hm, would it be better to log failure and carry on iteration instead of early terminating here? 

Concern is a single corrupt block on a single host would stop it's peers from being able to bootstrap from it.

Reply:
@robskillington Would the peer bootstrapping logic on the client side handle this situation properly if we just skipped over an entire block?

Reply:
We definitely shouldn't skip over an entire block, otherwise we magically did not hold up our deal of applying bootstrap level consistency (because one replica just looks like it was successful retrieving all metadata, even though it skipped some).

There's already a ton of places in this flow where we can return an error.  I'd prefer that we stick with that, in that case basically one replica will just look bad.  Depending on your bootstrap level consistency that may be fine (you might still get majority from the other nodes).  If not, an operator needs to make the call to reduce bootstrap consistency, or remove the problematic files themselves.

Reply:
There's logging on the client side for every error case thankfully, is that sufficient?